### PR TITLE
test: reenable main tests

### PR DESCRIPTION
**Pull request by kruskall**
main_test was disabled after the global state for the logsapi
listener was removed.
The new approach is to choose a random port and pass the address
to the app.
Add a timeout to make sure tests don't hang.

Blocked by https://github.com/elastic/apm-aws-lambda/issues/249

Closes https://github.com/elastic/apm-aws-lambda/issues/249

### CODE REVIEW

File: apm-lambda-extension/main_test.go [hunk 1]
```diff
@@ -502,191 +540,255 @@ func TestAPMServerRecovery(t *testing.T) {
...
+	}()
+
+	return ctx.Done()
+}
+
+func randomAddr() string {
+	// we cannot return a port that is already in use or it
+	// would return an error when creating the server.
+	// The solution is to spawn a test server to get a random
+	// port and immediately close it so that we can use the port.
```
Comment (axw) [inline apm-lambda-extension/main_test.go:788]: This is racy though, as another process could also be trying to acquire an ephemeral port. If the test ends up being flaky,  we may need to have the extension listen on an ephemeral port, log it, and then extract the listening port from the logs. Or something like that.

**Review by axw — APPROVED** [review]
Looks great!

File: apm-lambda-extension/main_test.go (hunk 1 above)
Comment (kruskall) [reply apm-lambda-extension/main_test.go:788]: Agree, the window to hit the race is small enough that I think it's safe to ignore. Another solution I had in mind was to use unix sockets.

**Pull request merged**